internal/controller/sqljob_controller.go

```diff
@@ -128,6 +128,7 @@ func (r *SqlJobReconciler) waitForDependencies(ctx context.Context, sqlJob *v1al
 			if err := r.patchStatus(ctx, sqlJob, r.ConditionComplete.PatcherFailed(msg)); err != nil {
 				return false, ctrl.Result{}, err
 			}
+			return false, ctrl.Result{}, err
```

Comment:
This is no longer needed, see the suggested solution here:
https://github.com/mariadb-operator/mariadb-operator/pull/924#pullrequestreview-2362478989